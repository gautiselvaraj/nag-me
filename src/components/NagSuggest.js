import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from './Button';
import Icon from './Icon';
import H5 from './H5';
import H6 from './H6';
import { roundedTimestamp, nextNagTimestamp } from '../utils/time';

const SuggestNagsHeading = styled.div`
  border-top: 1px solid ${props => props.theme.greyLighter};
  margin-top: 20px;
  padding-top: 10px;
  text-align: center;
`;

const SuggestNagsUl = styled.ul`
  list-style-type: none;
  margin: 0;
  overflow: hidden;
  padding: 0.5rem 0.5rem 1rem;
`;

const SuggestNagsLi = styled.li`
  background-color: ${props => props.theme.white};
  border-radius: 4px;
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.25);
  margin-bottom: 0.5rem;
  padding: 10px;
  padding-right: 50px;
  position: relative;
  user-select: none;
  text-align: left;

  &:last-child {
    margin-bottom: 0;
  }
`;

const SuggestNagsRepeat = styled.p`
  font-size: 0.8rem;
  margin: 5px 0 0;
`;

const SuggestNagsButton = styled.div`
  bottom: 0;
  position: absolute;
  right: 10px;
  top: 10px;
`;

export default class NagForm extends Component {
  static propTypes = {
    title: PropTypes.string,
    nagCreate: PropTypes.func.isRequired
  };

  state = {
    nags: [
      {
        title: 'Blink, relax your eyes',
        repeats: '15 min'
      },
      {
        title: 'Take a sip of water',
        repeats: '20 min'
      },
      {
        title: 'Take a deep breath',
        repeats: '1 hour'
      },
      {
        title: 'See a video and smile',
        repeats: '2 hour'
      },
      {
        title: 'Take a walk',
        repeats: '4 hour'
      }
    ]
  };

  handleNagCreate = (e, { title, repeats }) => {
    const on = nextNagTimestamp(roundedTimestamp(), repeats);
    this.props.nagCreate({
      title,
      firstNag: on,
      nextNag: on,
      repeats
    });
  };

  render() {
    return (
      <div>
        <SuggestNagsHeading>
          <H5>{this.props.title || 'Or start with nags below'}</H5>
        </SuggestNagsHeading>
        <SuggestNagsUl>
          {this.state.nags.map(nag => (
            <SuggestNagsLi key={nag.title}>
              <H6>{nag.title}</H6>
              <SuggestNagsRepeat>repeats every {nag.repeats}</SuggestNagsRepeat>
              <SuggestNagsButton>
                <Button
                  circle
                  title="Add this nag"
                  onClick={e => this.handleNagCreate(e, nag)}
                >
                  <Icon inverse add />
                </Button>
              </SuggestNagsButton>
            </SuggestNagsLi>
          ))}
        </SuggestNagsUl>
      </div>
    );
  }
}
