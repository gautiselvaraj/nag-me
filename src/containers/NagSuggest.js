import { connect } from 'react-redux';
import NagSuggest from '../components/NagSuggest';
import { nagCreate } from '../actions/NagActions';

const mapDispatchToProps = dispatch => ({
  nagCreate: nag => dispatch(nagCreate(nag))
});

export default connect(null, mapDispatchToProps)(NagSuggest);
