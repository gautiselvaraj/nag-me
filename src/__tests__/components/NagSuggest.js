import React from 'react';
import { mount } from 'enzyme';
import { fromJS } from 'immutable';
import NagSuggest from '../../components/NagSuggest';
import 'jest-styled-components';

describe('<NagSuggest />', () => {
  let nagCreate, originalSetTimeout;

  beforeEach(() => {
    nagCreate = jest.fn();
  });

  beforeAll(() => {
    originalSetTimeout = window.setTimeout;
    window.setTimeout = (callback, time) => callback();
  });

  afterAll(() => {
    window.setTimeout = originalSetTimeout;
  });

  it('should match snapshot and have correct elements and children', () => {
    const nagSuggest = mount(<NagSuggest nagCreate={nagCreate} />);

    expect(nagSuggest).toMatchSnapshot();
    expect(nagSuggest.find('h5').text()).toBe('Or try the nags below');
    expect(nagSuggest.find('li').length).toBe(5);
    nagSuggest
      .find('button')
      .first()
      .simulate('click');
    expect(nagCreate.mock.calls.length).toBe(1);
  });
});
