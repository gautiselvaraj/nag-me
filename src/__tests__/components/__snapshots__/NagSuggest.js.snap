// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NagSuggest /> should match snapshot 1`] = `
.c0 {
  border-top: 1px solid;
  margin-top: 20px;
  padding-top: 10px;
  text-align: center;
}

.c1 {
  font-size: 1rem;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.c2 {
  list-style-type: none;
  margin: 0;
  overflow: hidden;
  padding: 0.5rem 0.5rem 1rem;
}

.c3 {
  border-radius: 4px;
  box-shadow: 0 0 2px rgba(0,0,0,0.25);
  margin-bottom: 0.5rem;
  padding: 10px;
  padding-right: 50px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
}

.c3:last-child {
  margin-bottom: 0;
}

.c4 {
  font-size: 0.95rem;
  margin-top: 0.15rem;
  margin-bottom: 0.15rem;
}

.c5 {
  font-size: 0.8rem;
  margin: 5px 0 0;
}

.c6 {
  bottom: 0;
  position: absolute;
  right: 10px;
  top: 10px;
}

.c7 {
  border: 1px solid undefined;
  cursor: pointer;
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-radius: 50%;
  height: 2rem;
  text-align: center;
  width: 2rem;
}

.c7 i {
  position: relative;
  top: 2px;
}

.c8 {
  color: undefined;
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c8:before {
  content: "\\ea09";
}

<div>
  <div
    className="c0"
  >
    <h5
      className="c1"
    >
      Or try the nags below
    </h5>
  </div>
  <ul
    className="c2"
  >
    <li
      className="c3"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <h6
        className="c4"
      >
        Drink water
      </h6>
      <p
        className="c5"
      >
        repeats every 
        20 min
      </p>
      <div
        className="c6"
      >
        <button
          className="c7"
          onClick={[Function]}
          title="Start this nag"
        >
          <i
            className="c8"
          />
        </button>
      </div>
    </li>
    <li
      className="c3"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <h6
        className="c4"
      >
        Relax your eyes
      </h6>
      <p
        className="c5"
      >
        repeats every 
        30 min
      </p>
      <div
        className="c6"
      >
        <button
          className="c7"
          onClick={[Function]}
          title="Start this nag"
        >
          <i
            className="c8"
          />
        </button>
      </div>
    </li>
    <li
      className="c3"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <h6
        className="c4"
      >
        Take a deep breath
      </h6>
      <p
        className="c5"
      >
        repeats every 
        1 hour
      </p>
      <div
        className="c6"
      >
        <button
          className="c7"
          onClick={[Function]}
          title="Start this nag"
        >
          <i
            className="c8"
          />
        </button>
      </div>
    </li>
    <li
      className="c3"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <h6
        className="c4"
      >
        Remember to smile
      </h6>
      <p
        className="c5"
      >
        repeats every 
        4 hour
      </p>
      <div
        className="c6"
      >
        <button
          className="c7"
          onClick={[Function]}
          title="Start this nag"
        >
          <i
            className="c8"
          />
        </button>
      </div>
    </li>
    <li
      className="c3"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <h6
        className="c4"
      >
        Go for a walk
      </h6>
      <p
        className="c5"
      >
        repeats every 
        8 hour
      </p>
      <div
        className="c6"
      >
        <button
          className="c7"
          onClick={[Function]}
          title="Start this nag"
        >
          <i
            className="c8"
          />
        </button>
      </div>
    </li>
  </ul>
</div>
`;
